src/overlays/actors/ovl_Boss_Mo/z_boss_mo.c

```diff
@@ -22,299 +128,2943 @@ const ActorInit Boss_Mo_InitVars = {
     (ActorFunc)BossMo_Draw,
 };
 
-static ColliderCylinderInit D_8092603C = {
-    {
-        COLTYPE_HIT3,
-        AT_ON | AT_TYPE_ENEMY,
-        AC_ON | AC_TYPE_PLAYER,
-        OC1_ON | OC1_TYPE_PLAYER,
-        OC2_TYPE_1,
-        COLSHAPE_CYLINDER,
-    },
-    {
-        ELEMTYPE_UNK0,
-        { 0xFFCFFFFF, 0x00, 0x10 },
-        { 0xFFDFFFFF, 0x00, 0x00 },
-        TOUCH_ON | TOUCH_SFX_NORMAL,
-        BUMP_ON | BUMP_HOOKABLE,
-        OCELEM_ON,
-    },
-    { 20, 40, -20, { 0, 0, 0 } },
+static BossMo* sMorphaCore = NULL;
+static BossMo* sMorphaTent1 = NULL;
+static BossMo* sMorphaTent2 = NULL;
+
+static f32 sFlatWidth[41] = {
+    15.0f, 12.0f, 9.0f, 6.5f, 4.8f, 4.0f, 3.4f, 3.1f, 3.0f, 3.1f, 3.2f, 3.4f, 3.6f, 3.8f,
+    4.0f,  4.6f,  5.1f, 5.5f, 6.1f, 6.6f, 7.3f, 7.7f, 8.4f, 8.5f, 8.7f, 8.8f, 8.8f, 8.7f,
+    8.6f,  8.3f,  8.2f, 8.1f, 7.2f, 6.7f, 5.9f, 4.9f, 2.7f, 0.0f, 0.0f, 0.0f, 0.0f,
+};
+
+#include "z_boss_mo_colchk.c"
+
+static BossMoEffect sEffects[300];
+static s32 sSeed1;
+static s32 sSeed2;
+static s32 sSeed3;
+
+void BossMo_InitRand(s32 seedInit0, s32 seedInit1, s32 seedInit2) {
+    sSeed1 = seedInit0;
+    sSeed2 = seedInit1;
+    sSeed3 = seedInit2;
+}
+
+f32 BossMo_RandZeroOne(void) {
+    // Wichmann-Hill algorithm
+    f32 randFloat;
+
+    sSeed1 = (sSeed1 * 171) % 30269;
+    sSeed2 = (sSeed2 * 172) % 30307;
+    sSeed3 = (sSeed3 * 170) % 30323;
+
+    randFloat = (sSeed1 / 30269.0f) + (sSeed2 / 30307.0f) + (sSeed3 / 30323.0f);
+    while (randFloat >= 1.0f) {
+        randFloat -= 1.0f;
+    }
+    return fabsf(randFloat);
+}
+
+s32 BossMo_NearLand(Vec3f* pos, f32 margin) {
+    if (450.0f - margin <= fabsf(pos->x)) {
+        return true;
+    }
+    if (450.0f - margin <= fabsf(pos->z)) {
+        return true;
+    }
+    if ((fabsf(pos->x - 180.0f) < 90.0f + margin) || (fabsf(pos->x - -180.0f) < 90.0f + margin)) {
+        if (fabsf(pos->z - 180.0f) < 90.0f + margin) {
+            return true;
+        }
+        if (fabsf(pos->z - -180.0f) < 90.0f + margin) {
+            return true;
+        }
+    }
+    return false;
+}
+
+void BossMo_SpawnRipples(BossMoEffect* effect, Vec3f* pos, f32 scale, f32 vMaxScale, s16 maxAlpha, s16 partLimit,
```

Comment:
Using `vMaxScale` here seems confusing, was it intended?
Also the function only spawns 1 effect so it should probably be named singular to avoid any potential confusion.